WebGL related part GraphicsDevice is moving to a separate class

mvaligursky opened the pull request:
Related to https://github.com/playcanvas/engine/issues/3986

A base GraphicsDevice class is going to be platform independent, and WeblGraphicsDevice will contain WebGL specific implementation. In this first step, the whole implementation was simply moved to WeblGraphicsDevice.

slimbuck commented:
Have you tried using `git mv graphics-device.js webgl/blah` ? Might keep better history this way.

mvaligursky commented:
> Have you tried using `git mv graphics-device.js webgl/blah` ? Might keep better history this way.

I did, `git mv` and few different ways. Some of them work if you only move / rename file. But as soon as you modify it, the operation changes to delete/create pair. The suggestion online is to move / rename in a separate PR to the modify. But we cannot move files without modifying .. at least import paths.
I even created a test PR with two commits, first to move, second to change .. 
https://github.com/playcanvas/engine/pull/4007
when you inspect commits, there's perfect, the one with changes only shows small changes. But when you look at changed files for whole PR, the story is different :(

mvaligursky commented:
I'm also trying to split it into two separate PRs .. maybe that's a better way. Here: #4009 
I moved graphics-device.js to webgl subfolder .. and it's seen as move, even when I fixed the import paths. So this might preserve the history better. So maybe I do this as a first step, and then do follow up refactor after #4009 is merged.
